SpatialGDK/Source/SpatialGDKTests/Private/LocalDeploymentManager/LocalDeploymentManagerTest.cpp

```diff
@@ -0,0 +1,200 @@
+	static const FString AutomationLaunchConfig = TEXT("Improbable/AutomationLaunchConfig.json");
+
+	FLocalDeploymentManager* GetLocalDeploymentManager()
+	{
+		FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+		FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+		return LocalDeploymentManager;
+	}
+
+	FSpatialLaunchConfigDescription GenerateLaunchConfigDescription()
+	{
+		FSpatialLaunchConfigDescription LaunchConfigDescription;
+		FWorkerTypeLaunchSection UnrealWorkerDefaultSetting;
+		UnrealWorkerDefaultSetting.WorkerTypeName = FName(AutomationWorkerType);
+		UnrealWorkerDefaultSetting.Rows = 1;
+		UnrealWorkerDefaultSetting.Columns = 1;
+		UnrealWorkerDefaultSetting.bManualWorkerConnectionOnly = true;
+		LaunchConfigDescription.ServerWorkers.Reset();
+		LaunchConfigDescription.ServerWorkers.Add(UnrealWorkerDefaultSetting);
+
+		return LaunchConfigDescription;
+	}
+
+	bool GenerateWorkerPB()
+	{
+		FString BuildConfigArgs = TEXT("worker build build-config");
+		FString WorkerBuildConfigResult;
+		int32 ExitCode;
+		const FString SpatialExe(TEXT("spatial.exe"));
+		FSpatialGDKServicesModule::ExecuteAndReadOutput(SpatialExe, BuildConfigArgs, FSpatialGDKServicesModule::GetSpatialOSDirectory(), WorkerBuildConfigResult, ExitCode);
+
+		const int32 ExitCodeSuccess = 0;
+		return (ExitCode == ExitCodeSuccess);
+	}
+
+	bool GenerateWorkerJson()
+	{
+		const FString WorkerJsonDir = FSpatialGDKServicesModule::GetSpatialOSDirectory(TEXT("workers/unreal"));
+
+		FString JsonPath = FPaths::Combine(WorkerJsonDir, TEXT("spatialos.UnrealAutomation.worker.json"));
+		if (!FPaths::FileExists(JsonPath))
+		{
+			bool bRedeployRequired = false;
+			return GenerateDefaultWorkerJson(JsonPath, AutomationWorkerType.ToString(), bRedeployRequired);
+		}
+
+		return true;
+	}
+}
+
+DEFINE_LATENT_COMMAND(StartDeployment)
+{
+	if (!GenerateWorkerJson())
+	{
+		return true;
+	}
+
+	if (!GenerateWorkerPB())
+	{
+		return true;
+	}
+
+	const FString LaunchConfig = FPaths::Combine(FPaths::ConvertRelativePathToFull(FPaths::ProjectIntermediateDir()), AutomationLaunchConfig);
+	FSpatialLaunchConfigDescription LaunchConfigDescription = GenerateLaunchConfigDescription();
+	if (!GenerateDefaultLaunchConfig(LaunchConfig, &LaunchConfigDescription))
+	{
+		return true;
+	}
+
+	FLocalDeploymentManager* LocalDeploymentManager = GetLocalDeploymentManager();
+	if (LocalDeploymentManager->IsLocalDeploymentRunning())
+	{
+		return true;
+	}
+
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+	const FString LaunchFlags = SpatialGDKSettings->GetSpatialOSCommandLineLaunchFlags();
+
+	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [LocalDeploymentManager, LaunchConfig, LaunchFlags]
+	{
+		return LocalDeploymentManager->TryStartLocalDeployment(LaunchConfig, LaunchFlags);
```

Comment:
Nothing is listening for this return value, might be better to omit it entirely

Reply:
Removed return in start/stop functions' async tasks.